src/sentry/tasks/store.py

```diff
@@ -593,6 +593,12 @@ def _do_save_event(
                 # so we can delete it from the cache now.
                 if cache_key:
                     processing_store.delete_by_key(cache_key)
+                    if consumer_type == ConsumerType.Transactions:
```

Comment:
see line 588. this condition is already satisfied. you do not need to check it again.